src/Build/Evaluation/Evaluator.cs

```diff
@@ -1790,8 +1791,53 @@ private List<ProjectRootElement> ExpandAndLoadImports(string directoryOfImportin
                 if (solutionPath == "*Undefined*") solutionPath = null;
                 var projectPath = _data.GetProperty(ReservedPropertyNames.projectFullPath)?.EvaluatedValue;
 
+                var compareInfo = CultureInfo.InvariantCulture.CompareInfo;
+
+                static bool HasProperty(string value, CompareInfo compareInfo) =>
+                    value != null && compareInfo.IndexOf(value, "$(") != -1;
+
+                if (HasProperty(sdkReference.Name, compareInfo) ||
+                    HasProperty(sdkReference.Version, compareInfo) ||
+                    HasProperty(sdkReference.MinimumVersion, compareInfo))
+                {
+                    static string EvaluateProperty(string value, IElementLocation location, Expander<P, I> expander) =>
+                        value != null
+                            ? expander.ExpandIntoStringAndUnescape(value,
+                                                                   ExpanderOptions.ExpandProperties,
+                                                                   location)
+                            : null;
+
+                    var sdkReferenceOrigin = importElement.SdkLocation;
+
+                    var expandedSdkReference = new SdkReference(
+                        EvaluateProperty(sdkReference.Name, sdkReferenceOrigin, _expander),
+                        EvaluateProperty(sdkReference.Version, sdkReferenceOrigin, _expander),
+                        EvaluateProperty(sdkReference.MinimumVersion, sdkReferenceOrigin, _expander)
+                    );
+
+                    if (!Equals(expandedSdkReference, sdkReference))
+                    {
+                        var eventArgs = new ProjectImportedEventArgs(
```

Comment:
AFAIK msbuild does not log expansions, in which case it's probably best to be consistent and not log this.s

Reply:
But MSBuild logs e.g. property overrides, which really helps debug cases when something expected is overridden with sth unexpected. I can remove this, of course (that would also drop the new string in resx), but I tried to make this a bit more debuggable.

Reply:
I agree with the debuggability part, however the following arguments make me not want the log event for this case:
- other expansions are not logged. If we decide to do something else, which I agree might improve debuggability, I'd rather we have a unified approach that logs all expansions. That's a bigger project for a different PR, and needs careful design to avoid perf issues or log size inflation.
- the expanded sdk reference information already kinda gets indirectly logged when the import on the expanded sdk reference gets done. There is room for improvement here: the ProjectImportedEventArgs does not contain information on whether the import was an sdk import and if it was, what the SdkReference contents was. It's a somewhat unrelated change, if you do it you can open a new PR with it, rebased on this current PR, and we'll merge them in cadence. But I think it would be super useful for debugging to be able to search the binlog for all the sdk based imports.